tests/component/component.test.ts

```diff
@@ -149,6 +149,25 @@ describe("basic widget properties", () => {
     expect(fixture.innerHTML).toBe("<div>1<button>Inc</button></div>");
   });
 
+  test("support for callable expression in event handler", async () => {
+    class Counter extends Component {
+      static template = xml`
+      <div><t t-esc="state.counter"/><button t-on-click="obj.onClick">Inc</button></div>`;
+      state = useState({
+        counter: 0,
+      });
+      obj = { onClick: ev => this.state.counter++ }
```

Comment:
This might not be the best test to cover it, as it doesn't make use of `ev` here. If you are ok with the proposition I guess I should do one with input and reading value of target.